server/src/com/cloud/storage/VolumeApiServiceImpl.java

```diff
@@ -1835,6 +1847,25 @@ private Volume orchestrateDetachVolumeFromVM(long vmId, long volumeId) {
         }
     }
 
+    public void updateMissingRootDiskController(final VMInstanceVO vm, final String rootVolChainInfo) {
+        if (vm != null && vm.getType().equals(VirtualMachine.Type.User) && !Strings.isNullOrEmpty(rootVolChainInfo)) {
```

Comment:
very nice the extraction you did here.
What about the short-circuit for the condition at line 1851?

```
if (vm == null || !VirtualMachine.Type.User.equals(vm.getType()) || Strings.isNullOrEmpty(rootVolChainInfo)) {
	return;
}
...
the rest of the code (lines 1852-1865)
...

```